### PR TITLE
ASR: A2A support, V2A and B2A fixes, HUB changes.

### PR DESCRIPTION
**Pull request by avneeshrai**
1) Changes to support A2A provider operations.
2) V2A scenario fixes.
3) HUB changes
4) B2A Exclude Disk Contract Changes.
5) Changes for job status event

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@avneeshrai__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (avrai). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by markcowl** [comment]
@avneeshrai Please respond to review comments here and regenerate:  https://github.com/Azure/hydra-specs-pr/pull/1341

**Comment by avneeshrai** [comment]
@markcowl have addressed your comments and updated the PR after bumping up the major library version.

File: src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs [hunk 1]
```diff
@@ -28,7 +29,6 @@ public class SiteTests : SiteRecoveryTestsBase
     {
         string siteName = "site3";
 
-        [Fact]
```
Comment (markcowl) [inline src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs]: Why remove this test?

File: src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs [hunk 2]
```diff
@@ -47,7 +46,6 @@ public void CreateSite()
             }
         }
 
-        [Fact]
```
Comment (markcowl) [inline src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs]: Same question

**Comment by markcowl** [comment]
@avneeshrai Please add test coverage for the new operations

File: src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs (hunk 1 above)
Comment (avneeshrai) [inline src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs]: It was unintended removal. Added back.

File: src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs (hunk 2 above)
Comment (avneeshrai) [inline src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs]: It was unintended removal. Added back.

**Comment by avneeshrai** [comment]
@markcowl We have a dedicated test infra for test coverage that covers all our scenarios and operations.
Adding here too would be duplication of efforts for us with no plus. 
Let me know if more details are needed. 

Updated the PR after re-adding create and remove site tests.

**Comment by markcowl** [comment]
@avneeshrai You need to have tests that cover the new scenarios here.  Since this is a public repo, the tests that someone would need to run to verify contributions need to be here.

**Comment by avneeshrai** [comment]
@markcowl Added tests.

File: src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs [hunk 3]
```diff
@@ -60,6 +62,40 @@ public void DeleteSite()
...
+
+                FabricCreationInput siteInput = new FabricCreationInput();
+                siteInput.Properties = new FabricCreationInputProperties();
+
+                siteInput.Properties.CustomDetails = new AzureFabricCreationInput()
+                {
+                    Location = location
+                };
+
+                var response = client.Fabrics.BeginCreating(azureSiteName, siteInput, RequestHeaders);
```
Comment (markcowl) [inline src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs]: WHay are you not ensuring that the fabric is created?

File: src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs [hunk 4]
```diff
@@ -60,6 +62,40 @@ public void DeleteSite()
...
+
+        [Fact]
+        public void A2ADeleteSite()
+        {
+            using (UndoContext context = UndoContext.Current)
+            {
+                context.Start();
+                var client = GetSiteRecoveryClient(CustomHttpHandler, Constants.A2A);
+
+                var response = client.Fabrics.BeginDeleting(azureSiteName, RequestHeaders);
```
Comment (markcowl) [inline src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs]: Why are you not ensuring that deletion is successful?

File: src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs (hunk 3 above)
Comment (avneeshrai) [inline src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs]: This flow is testing the async call. So we are not expecting fabric to be created even after the call is a success.

File: src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs (hunk 4 above)
Comment (avneeshrai) [inline src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs]: This flow is testing the async call. So we are not expecting fabric to be deleted even after the call is a success.

**Comment by avneeshrai** [comment]
@markcowl Responded to your comments.

File: src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs (hunk 4 above)
Comment (avneeshrai) [inline src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs]: Unlike Policy deletion test which tests sync flow.

File: src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs (hunk 3 above)
Comment (avneeshrai) [inline src/ResourceManagement/SiteRecovery/SiteRecovery.Tests/ScenarioTests/SiteTests.cs]: Unlike Policy creation test which tests sync flow.

**Comment by avneeshrai** [comment]
@markcowl  Please let me know if you see some issues here.

**Pull request merged**